test/integration/content-explorer/ContentExplorer.e2e.test.js

```diff
@@ -0,0 +1,207 @@
+// <reference types="Cypress" />
+const helpers = {
+    load(additionalProps = { canUpload: true, permissions: { canUpload: true } }) {
+        cy.visit('/Elements/ContentExplorer', {
+            onBeforeLoad: contentWindow => {
+                contentWindow.PROPS = additionalProps;
+            },
+        });
+    },
+    getRow(rowNum) {
+        return cy.getByDataPreview('ContentExplorer').find(`.bce-item-row-${rowNum}`);
+    },
+    checkRows(rowNum) {
+        cy.getByDataPreview('ContentExplorer')
+            .find('.ReactVirtualized__Table')
+            .then(table => {
+                const totalRows = table[0].getAttribute('aria-rowcount');
+
+                for (let i = 0; i < totalRows; i += 1) {
+                    if (i !== rowNum) {
+                        this.getRow(i).should('not.have.class', 'bce-item-row-selected');
+                    }
+                }
+                if (rowNum >= 0) {
+                    this.getRow(rowNum).should('have.class', 'bce-item-row-selected');
+                }
+            });
+    },
+    selectRow(rowNum) {
+        this.getRow(rowNum)
+            .as('row')
+            .click()
+            .should('have.class', 'bce-item-row-selected');
+
+        // Whenever a row is selected, confirm that it is the only row that appears selected
+        this.checkRows(rowNum);
+        return cy.get('@row');
+    },
+    clickAddButton() {
+        cy.getByDataPreview('ContentExplorer')
+            .find('.be-btn-add')
+            .click();
+    },
+    clickUploadButton() {
+        cy.get('.dropdown-menu-element > ul > li')
+            .contains('Upload')
+            .click();
+    },
+    clickNewFolderButton() {
+        cy.get('.dropdown-menu-element > ul > li')
+            .contains('New Folder')
+            .click();
+    },
+    clickNewFolderCancelButton() {
+        cy.get('.be-modal-btns > button')
+            .contains('Cancel')
+            .click();
+    },
+    clickNewFolderCreateButton() {
+        cy.get('.be-modal-btns > button')
+            .contains('Create')
+            .click();
+    },
+    clickCloseUploadModal() {
+        cy.get('.bcu-footer-left > button')
```

Comment:
It's not usually necessary to combine `get` and `contains`, as the latter usually only matches a single item. We should also avoid using CSS selectors (and especially descendent element selectors) whenever possible in Cypress tests.